python/ql/test/experimental/dataflow/tainttracking/customSanitizer/TestTaint.ql

```diff
@@ -1,5 +1,19 @@
 import experimental.dataflow.tainttracking.TestTaintLib
 
+class IsSafeCheck extends DataFlow::BarrierGuard {
+  Variable v;
+
+  IsSafeCheck() {
+    this.(CallNode).getNode().getFunc().(Name).getId() = "emulated_is_safe" and
+    this.(CallNode).getAnArg().(NameNode).uses(v)
+  }
+
+  override predicate checks(ControlFlowNode node, boolean testIsTrue) {
+    node.(NameNode).uses(v) and
```

Comment:
The `node` should simply be `this.(CallNode).getAnArg()`. The matching of the two variable uses should be done inside `final Node getAGuardedNode()` in `BarrierGuard`

Reply:
That is a much nicer API :-)